### ISSUE TITLE
[Bug]: You are using a model of type qwen2_vl to instantiate a model of type . This is not supported for all configurations of models and can yield errors.

### ISSUE THREAD

**Issue by cqray1990** [vllm-project/vllm#9872]
### Your current environment

vllm=0.6.3

### Model Input Dumps

You are using a model of type qwen2_vl to instantiate a model of type . This is not supported for all configurations of models and can yield errors.

### 🐛 Describe the bug

vllm of 0.6.3 supoorted Qwen2-vl Now ,but when inerence , it raise an warning,did't it result to some errors?
You are using a model of type qwen2_vl to instantiate a model of type . This is not supported for all configurations of models and can yield errors.

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by DarkLight1337** [comment]
Can you show the full logs?

**Comment by cqray1990** [comment]
> Can you show the full logs?

You are using a deprecated `pynvml` package. Please install `nvidia-ml-py` instead, and make sure to uninstall `pynvml`. When both of them are installed, `pynvml` will take precedence and cause errors. See https://pypi.org/project/pynvml for more information.
/lib/python3.10/site-packages/vllm/connections.py:8: RuntimeWarning: Failed to read commit hash:
No module named 'vllm._version'
  from vllm.version import __version__ as VLLM_VERSION
You are using a model of type qwen2_vl to instantiate a model of type . This is not supported for all configurations of models and can yield errors.
INFO 10-31 15:39:22 llm_engine.py:237] Initializing an LLM engine (vdev) with config: model='/Qwen2-vl-2B-instruct', speculative_config=None, tokenizer='/Qwen2-vl-2B-instruct', skip_tokenizer_init=False, tokenizer_mode=auto, revision=None, override_neuron_config=None, rope_scaling=None, rope_theta=None, tokenizer_revision=None, trust_remote_code=False, dtype=torch.bfloat16, max_seq_len=4096, download_dir=None, load_format=LoadFormat.AUTO, tensor_parallel_size=1, pipeline_parallel_size=1, disable_custom_all_reduce=False, quantization=None, enforce_eager=False, kv_cache_dtype=auto, quantization_param_path=None, device_config=cuda, decoding_config=DecodingConfig(guided_decoding_backend='outlines'), observability_config=ObservabilityConfig(otlp_traces_endpoint=None, collect_model_forward_time=False, collect_model_execute_time=False), seed=0, served_model_name=/Qwen2-vl-2B-instruct, use_v2_block_manager=True, num_scheduler_steps=1, chunked_prefill_enabled=False multi_step_stream_outputs=True, enable_prefix_caching=False, use_async_output_proc=True, use_cached_outputs=False, mm_processor_kwargs=None)

**Comment by DarkLight1337** [comment]
What does your HF config look like?

**Comment by cqray1990** [comment]
> What does your HF config look like?

{
  "architectures": [
    "Qwen2VLForConditionalGeneration"
  ],
  "attention_dropout": 0.0,
  "bos_token_id": 151643,
  "eos_token_id": 151645,
  "vision_start_token_id": 151652,
  "vision_end_token_id": 151653,
  "vision_token_id": 151654,
  "image_token_id": 151655,
  "video_token_id": 151656,
  "hidden_act": "silu",
  "hidden_size": 1536,
  "initializer_range": 0.02,
  "intermediate_size": 8960,
  "max_position_embeddings": 32768,
  "max_window_layers": 28,
  "model_type": "qwen2_vl",
  "num_attention_heads": 12,
  "num_hidden_layers": 28,
  "num_key_value_heads": 2,
  "rms_norm_eps": 1e-06,
  "rope_theta": 1000000.0,
  "sliding_window": 32768,
  "tie_word_embeddings": true,
  "torch_dtype": "bfloat16",
  "transformers_version": "4.41.2",
  "use_cache": true,
  "use_sliding_window": false,
  "vision_config": {
    "depth": 32,
    "embed_dim": 1280,
    "mlp_ratio": 4,
    "num_heads": 16,
    "in_chans": 3,
    "hidden_size": 1536,
    "patch_size": 14,
    "spatial_merge_size": 2,
    "spatial_patch_size": 14,
    "temporal_patch_size": 2
  },
  "rope_scaling": {
    "type": "mrope",
    "mrope_section": [
      16,
      24,
      24
    ]
  },
  "vocab_size": 151936
}

**Comment by DarkLight1337** [comment]
Are you currently using vLLM built from source? Can you update to latest main and see if the problem still persists?

**Comment by cqray1990** [comment]
> Are you currently using vLLM built from source? Can you update to latest main and see if the problem still persists?

i didn't build from source, instead, by pip install vllm==0.6.3,
but
another question, when i build from source,  it need from vllm.vllm.XXX import XXX..., that's weird, normaly, it just from vllm.XXX import XXX   pip install -e .    from source,

**Comment by DarkLight1337** [comment]
Make sure you are inside the repo directory before running `pip install`

**Comment by cqray1990** [comment]
> pip install

i build as official said
git clone https://github.com/vllm-project/vllm.git
cd vllm
pip install -e .

**Comment by DarkLight1337** [comment]
Can you run `pip list` and show the output for vLLM?

**Comment by cqray1990** [comment]
> Can you run `pip list` and show the output for vLLM?

vllm        /home/user/vllm

**Comment by DarkLight1337** [comment]
Have you tried importing from `vllm` directly? It should work even when `pip install -e` is run at the root of the repo (where the source files are nested under `vllm` of the repo)/

**Comment by cqray1990** [comment]
> Have you tried importing from `vllm` directly? It should work even when `pip install -e` is run at the root of the repo (where the source files are nested under `vllm` of the repo).

it can't be import, only from vllm.vllm.XXX import XXX,  can you  build as official said , and test with from vllm.XXX import XXX, there is no problem by building in that way

**Comment by DarkLight1337** [comment]
> > Have you tried importing from `vllm` directly? It should work even when `pip install -e` is run at the root of the repo (where the source files are nested under `vllm` of the repo).
> 
> it can't be import, only from vllm.vllm.XXX import XXX, can you build as official said , and test with from vllm.XXX import XXX, there is no problem by building in that way

Yes, I am able to import `from vllm.XXX import YYY` as usual. Are you running python with a file named `vllm.py` in your current local directory?

**Comment by cqray1990** [comment]
no .



---- Replied Message ----
| From | Cyrus ***@***.***> |
| Date | 10/31/2024 18:46 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [vllm-project/vllm] [Bug]: You are using a model of type qwen2_vl to instantiate a model of type . This is not supported for all configurations of models and can yield errors. (Issue #9872) |

Have you tried importing from vllm directly? It should work even when pip install -e is run at the root of the repo (where the source files are nested under vllm of the repo).

it can't be import, only from vllm.vllm.XXX import XXX, can you build as official said , and test with from vllm.XXX import XXX, there is no problem by building in that way

Yes, I am able to import from vllm.XXX import YYY as usual. Are you running python with a file named vllm.py in your current local directory?

**Comment by cqray1990** [comment]
it is so wired



---- Replied Message ----
| From | Cyrus ***@***.***> |
| Date | 10/31/2024 18:46 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [vllm-project/vllm] [Bug]: You are using a model of type qwen2_vl to instantiate a model of type . This is not supported for all configurations of models and can yield errors. (Issue #9872) |

Have you tried importing from vllm directly? It should work even when pip install -e is run at the root of the repo (where the source files are nested under vllm of the repo).

it can't be import, only from vllm.vllm.XXX import XXX, can you build as official said , and test with from vllm.XXX import XXX, there is no problem by building in that way

Yes, I am able to import from vllm.XXX import YYY as usual. Are you running python with a file named vllm.py in your current local directory?
